hbase-common/src/main/java/org/apache/hadoop/hbase/io/crypto/tls/X509Util.java

```diff
@@ -0,0 +1,320 @@
+      keyStoreType = "jks";
+    }
+
+    try {
+      char[] password = keyStorePassword.toCharArray();
+      KeyStore ks = KeyStore.getInstance(keyStoreType);
+      try (InputStream inputStream = new BufferedInputStream(
+        Files.newInputStream(new File(keyStoreLocation).toPath()))) {
+        ks.load(inputStream, password);
+      }
+
+      KeyManagerFactory kmf = KeyManagerFactory.getInstance("PKIX");
+      kmf.init(ks, password);
+
+      for (KeyManager km : kmf.getKeyManagers()) {
+        if (km instanceof X509KeyManager) {
+          return (X509KeyManager) km;
+        }
+      }
+      throw new X509Exception.KeyManagerException("Couldn't find X509KeyManager");
+    } catch (IOException | GeneralSecurityException | IllegalArgumentException e) {
+      throw new X509Exception.KeyManagerException(e);
+    }
+  }
+
+  /**
+   * Creates a trust manager by loading the trust store from the given file
+   * of the given type, optionally decrypting it using the given password.
+   * @param trustStoreLocation the location of the trust store file.
+   * @param trustStorePassword optional password to decrypt the trust store
+   *                           (only applies to JKS trust stores). If empty,
+   *                           assumes the trust store is not encrypted.
+   * @param trustStoreType     must be JKS, PEM, PKCS12, BCFKS or null. If
+   *                           null, attempts to autodetect the trust store
+   *                           type from the file extension (e.g. .jks / .pem).
+   * @param crlEnabled enable CRL (certificate revocation list) checks.
+   * @param ocspEnabled enable OCSP (online certificate status protocol)
+   *                    checks.
+   * @return the trust manager.
+   * @throws X509Exception.TrustManagerException if something goes wrong.
+   */
+  public static X509TrustManager createTrustManager(
+    String trustStoreLocation,
+    String trustStorePassword,
+    String trustStoreType,
+    boolean crlEnabled,
+    boolean ocspEnabled) throws
+    X509Exception.TrustManagerException {
+
+    if (trustStorePassword == null) {
+      trustStorePassword = "";
+    }
+
+    if (trustStoreType == null) {
+      trustStoreType = "jks";
+    }
+
+    try {
+      char[] password = trustStorePassword.toCharArray();
+      KeyStore ts = KeyStore.getInstance(trustStoreType);
+      try (InputStream inputStream = new BufferedInputStream(
+        Files.newInputStream(new File(trustStoreLocation).toPath()))) {
+        ts.load(inputStream, password);
+      }
+
+      PKIXBuilderParameters pbParams = new PKIXBuilderParameters(ts, new X509CertSelector());
+      if (crlEnabled || ocspEnabled) {
+        pbParams.setRevocationEnabled(true);
+        System.setProperty("com.sun.net.ssl.checkRevocation", "true");
+        System.setProperty("com.sun.security.enableCRLDP", "true");
+        if (ocspEnabled) {
+          Security.setProperty("ocsp.enable", "true");
+        }
```

Comment:
Do we have to set these via system property to be picked up by the javax code? These are also specific to Oracle JDK -- IBM and others may have different properties. Fine to focus on Oracle jdk for now.

Reply:
ocsp is common, yeah, but `com.ibm.security.enableCRLDP` and `com.ibm.jsse2.checkRevocation` look like the equivalent properties for IBM java. Deal with this later.